packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -561,25 +568,58 @@ export abstract class FluidDataStoreContext
 		this.channel!.setConnectionState(connected, clientId);
 	}
 
-	public process(
-		message: ISequencedDocumentMessage,
+	/**
+	 * Process the messages to maintain backwards compatibility. The `processMessages` function is added to
+	 * the channel in 2.4.0. For channels before that, call `process` for each message.
+	 */
+	private processMessagesCompat(
+		channel: IFluidDataStoreChannel,
+		message: ISequencedRuntimeMessageCore,
+		messageContents: IRuntimeMessageContents[],
+		local: boolean,
+	) {
+		if (channel.processMessages !== undefined) {
```

Comment:
I'd add some //back-compat comments and link to ADO bug to cleanup any code that is temporary.

Reply:
Added back-compat comment and created an ADO item for cleanup.